### PR TITLE
fix #78743 tennisassa.fi

### LINKED ISSUE #78743: www.tennisassa.fi
### Issue URL (Ads)
 [https://www.tennisassa.fi/cookiewall/](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Fwww.tennisassa.fi%2Fcookiewall%2F)

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  Screenshot 1
</details>

### System configuration

Information | value
--- | ---
AdGuard product: | AdGuard for Windows v7.6 nightly 27
Browser: | Chrome
AdGuard driver: | WFP
Stealth mode options: | Hide your search queries, <br>Send Do-Not-Track header, <br>Remove X-Client-Data header from HTTP requests, <br>Strip URLs from tracking parameters, <br>Self-destructing third-party cookies (180), <br>Block Push API, <br>Block Location API
DNS filtering: | server: `https://dns-unfiltered.adguard.com/dns-query`<br>filters: `User rules`
Filters: | <b>Ad Blocking:</b><br/> EasyList, <br/>Peter Lowe's Blocklist, <br/>AdGuard Mobile Ads, <br/>AdGuard Base<br/><br/><b>Privacy:</b><br/> EasyPrivacy, <br/>AdGuard Tracking Protection<br/><br/><b>Social Widgets:</b><br/> AdGuard Social Media<br/><br/><b>Annoyances:</b><br/> Adblock Warning Removal List, <br/>I don't care about cookies, <br/>EasyList Cookie List, <br/>AdGuard Annoyances<br/><br/><b>Security:</b><br/> Online Malicious URL Blocklist, <br/>Spam404, <br/>NoCoin Filter List<br/><br/><b>Language-specific:</b><br/> Adblock List for Finland
Custom filters: | https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt,<br>https://raw.githubusercontent.com/DandelionSprout/adfilt/master/LegitimateURLShortener.txt
Userscripts: | https://userscripts.adtidy.org/beta/assistant/4.3/assistant.user.js,<br>https://userscripts.adtidy.org/beta/adguard-extra/1.0/adguard-extra.user.js

### PR DESCRIPTION
**Pull request by wout-codes**
#78743

### CODE REVIEW

File: AnnoyancesFilter/sections/cookies_specific.txt [hunk 1]
```diff
@@ -5358,6 +5358,7 @@ selgros.pl#%#//scriptlet('set-cookie', 'cookie_new_msg', '1')
 guard.io#%#//scriptlet('set-cookie', 'cookie_consent_seen', '1')
 consent.youtube.com#%#AG_onLoad(function(){if(window.location.href.includes("consent.youtube.com")){var b=new MutationObserver(function(){var a=document.querySelector('form[action="https://consent.youtube.com/s"] > input[type="submit"]');a&&a.click();console.log('sadsa');});b.observe(document,{childList:!0,subtree:!0});setTimeout(function(){b.disconnect()},1E4)}});
 meyer-gartenbedarf.de#%#//scriptlet('set-cookie', 'amcookie_allowed', '1')
+tennisassa.fi#%#AG_onLoad(function() { setTimeout(function() { var el = document.querySelector('form[action="/cookiewall/Save"] > .js-simple-accept-view > button[type="submit"]'); if(el) { el.click(); } }, 300); });
```
Comment (zloyden) [inline AnnoyancesFilter/sections/cookies_specific.txt:5361]: Better use `MutationObserver` with a conditional.
```
tennisassa.fi#%#AG_onLoad(function(){if(location.pathname === '/cookiewall/'){var a=new MutationObserver(function(){var b=document.querySelector(".js-accept-basic");b&&(a.disconnect(),b.click())});a.observe(document,{childList:!0,subtree:!0});setTimeout(function(){a.disconnect()},1E4)}});
```

**Review by zloyden — CHANGES_REQUESTED** [review]

File: EnglishFilter/sections/foreign.txt [hunk 2]
```diff
@@ -2996,6 +2996,8 @@ kava.ee##.kava_ad_chilli
 !------------------------------ Finnish ------------------------------
 !---------------------------------------------------------------------
 !
+! https://github.com/AdguardTeam/AdguardFilters/issues/78743
+tennisassa.fi#?#.module__content:has(> div[class] > .ad--container)
```
Comment (zloyden) [inline EnglishFilter/sections/foreign.txt:3000]: I guess you want to block leftover, but this rule doesn't do it properly.
`tennisassa.fi#$?#.page-content__main > div.module > div.module__content div.ad--container:upward(div.module) { remove: true; }`

**Review by zloyden — CHANGES_REQUESTED** [review]

File: AnnoyancesFilter/sections/cookies_specific.txt (hunk 1 above)
Comment (wout-codes) [reply AnnoyancesFilter/sections/cookies_specific.txt:5361]: I don't think `MutationObserver` is needed in this case.

Comment (wout-codes) [reply AnnoyancesFilter/sections/cookies_specific.txt:5361]: @zloyden I have also tried a similar rule before adding a simpler rule. This rule doesn't click the Accept Cookies at first visit, it works fine after refreshing the site.

Comment (zloyden) [reply AnnoyancesFilter/sections/cookies_specific.txt:5361]: > I don't think `MutationObserver` is needed in this case.

Why so? Your rule will apply to the entire part of the website, you also depend on a delay of `300`.


> I have also tried a similar rule before adding a simpler rule. This rule doesn't click the Accept Cookies at first visit, it works fine after refreshing the site.

Try without `AG_onLoad`

```
tennisassa.fi#%#(function(){if("/cookiewall/"===location.pathname){var a=new MutationObserver(function(){var b=document.querySelector(".js-accept-basic");b&&(a.disconnect(),b.click())});a.observe(document,{childList:!0,subtree:!0});setTimeout(function(){a.disconnect()},1E4)}})();
```

Comment (wout-codes) [reply AnnoyancesFilter/sections/cookies_specific.txt:5361]: It works now. Btw, why didn't it work with `AG_onLoad`?

**Review by zloyden — APPROVED** [review]

File: EnglishFilter/sections/foreign.txt [hunk 3]
```diff
@@ -2996,6 +2996,8 @@ kava.ee##.kava_ad_chilli
 !------------------------------ Finnish ------------------------------
 !---------------------------------------------------------------------
 !
+! https://github.com/AdguardTeam/AdguardFilters/issues/78743
+tennisassa.fi#$?#.page-content__main > div.module > div.module__content div.ad--container:upward(div.module) { remove: true; }
```
Comment (zloyden) [inline EnglishFilter/sections/foreign.txt:3000]: Please re-write this one a bit
`tennisassa.fi#$?#.page-content__main > div.module:first-child > div.module__content div.ad--container:upward(div.module) { remove: true; }`

**Review by zloyden — CHANGES_REQUESTED** [review]

**Review by AdamWr — APPROVED** [review]

File: AnnoyancesFilter/sections/cookies_specific.txt (hunk 1 above)
Comment (zloyden) [reply AnnoyancesFilter/sections/cookies_specific.txt]: When it's used with `AG_onLoad`, probably because there is not changes in DOM.

**Review by zloyden — APPROVED** [review]

**Pull request merged**